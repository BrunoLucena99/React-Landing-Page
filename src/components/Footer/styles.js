import styled from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  text-align: center;
  border-top: ${({ theme }) => `0.1rem solid ${theme.colors.mediumGray}`};

  a {
    color: inherit;
    text-decoration: none;
  }

  ${TextComponent} {
    font-size: ${({ theme }) => theme.font.sizes.small};
  }

  ${SectionContainer} {
    padding-top: 0;
    padding-bottom: 0;
  }
`;
