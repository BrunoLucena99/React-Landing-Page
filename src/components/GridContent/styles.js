import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const HTMLContainer = styled.div`
  margin: ${({ theme }) => theme.spacings.xhuge} 0;
`;
