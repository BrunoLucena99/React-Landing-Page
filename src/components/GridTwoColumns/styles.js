import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.xlarge};

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
