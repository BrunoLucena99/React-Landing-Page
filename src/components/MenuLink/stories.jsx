import MenuLink from '.';

export default {
  title: 'MenuLink',
  component: MenuLink,
  args: {
    children: `Inicio`,
    href: 'https://www.google.com/',
  },
  argTypes: {
    children: { type: 'string' },
    newTab: { type: 'boolean' },
    href: { type: 'string' },
  },
};

export const Template = (args) => (
  <div style={{ display: 'flex' }}>
    <MenuLink {...args} />
  </div>
);
