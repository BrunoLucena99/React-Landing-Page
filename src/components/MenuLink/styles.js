import styled from 'styled-components';

export const Container = styled.a`
  font-size: ${({ theme }) => theme.font.sizes.small};
  display: block;
  text-decoration: none;
  padding: ${({ theme }) => theme.spacings.small};
  color: ${({ theme }) => theme.colors.primaryColor};
  position: relative;

  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 50%;
    width: 0;
    height: 0.2rem;
    background: ${({ theme }) => theme.colors.secondaryColor};
    transition: all 300ms ease-in-out;
  }

  &:hover::after {
    left: 25%;
    width: 50%;
  }
`;
