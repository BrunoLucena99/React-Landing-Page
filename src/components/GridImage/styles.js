import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
