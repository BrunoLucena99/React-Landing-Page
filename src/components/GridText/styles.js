import styled from 'styled-components';
import { Title as HeadingTitle } from '../Heading/styles';

export const Container = styled.div``;

export const Grid = styled.div`
  counter-reset: grid-counter;
  display: grid;
  gap: ${({ theme }) => theme.spacings.large};
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  margin-top: ${({ theme }) => theme.spacings.xhuge};
`;

export const GridElement = styled.div`
  ${HeadingTitle} {
    position: relative;
    left: 5rem;
  }

  ${HeadingTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`;
