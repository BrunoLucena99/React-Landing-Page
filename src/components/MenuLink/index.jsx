import P from 'prop-types';
import { Container } from './styles';

const MenuLink = ({ children, href, newTab = false }) => {
  const target = newTab ? '_blank' : '_self';
  return (
    <Container href={href} target={target}>
      {children}
    </Container>
  );
};

MenuLink.propTypes = {
  children: P.node.isRequired,
  href: P.string.isRequired,
  newTab: P.bool,
};

export default MenuLink;
