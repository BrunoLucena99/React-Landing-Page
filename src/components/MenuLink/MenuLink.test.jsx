import { renderTheme } from '../../styles/RenderTheme';
import MenuLink from '.';
import { screen } from '@testing-library/react';

describe('<MenuLink />', () => {
  it('should render correctly', () => {
    renderTheme(<MenuLink href="#inicio">Inicio</MenuLink>);
    const element = screen.getByRole('link', { name: 'Inicio' });
    expect(element).toBeInTheDocument();
    expect(element).toHaveAttribute('target', '_self');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <MenuLink href="#inicio">Inicio</MenuLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should open link in a new tab', () => {
    renderTheme(
      <MenuLink href="#inicio" newTab>
        Inicio
      </MenuLink>,
    );
    const element = screen.getByRole('link', { name: 'Inicio' });
    expect(element).toHaveAttribute('target', '_blank');
  });
});
