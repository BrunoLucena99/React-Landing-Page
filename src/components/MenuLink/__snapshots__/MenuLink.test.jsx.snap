// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="#inicio"
  target="_self"
>
  Inicio
</a>
`;
